metadata-io/src/main/java/com/linkedin/metadata/timeline/eventgenerator/DataProcessInstanceRunEventChangeEventGenerator.java

```diff
@@ -0,0 +1,99 @@
+package com.linkedin.metadata.timeline.eventgenerator;
+
+import com.linkedin.common.AuditStamp;
+import com.linkedin.common.urn.Urn;
+import com.linkedin.dataprocess.DataProcessInstanceRelationships;
+import com.linkedin.dataprocess.DataProcessInstanceRunEvent;
+import com.linkedin.dataprocess.DataProcessRunStatus;
+import com.linkedin.metadata.entity.EntityService;
+import com.linkedin.metadata.timeline.data.ChangeCategory;
+import com.linkedin.metadata.timeline.data.ChangeEvent;
+import com.linkedin.metadata.timeline.data.ChangeOperation;
+import java.net.URISyntaxException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import static com.linkedin.metadata.Constants.*;
+
+
+public class DataProcessInstanceRunEventChangeEventGenerator
+    extends EntityChangeEventGenerator<DataProcessInstanceRunEvent> {
+  public DataProcessInstanceRunEventChangeEventGenerator(@Nonnull final EntityService entityService) {
+    super(entityService);
+  }
+
+  @Override
+  public List<ChangeEvent> getChangeEvents(@Nonnull Urn urn, @Nonnull String entity, @Nonnull String aspect,
+      @Nonnull Aspect<DataProcessInstanceRunEvent> from, @Nonnull Aspect<DataProcessInstanceRunEvent> to,
+      @Nonnull AuditStamp auditStamp) {
+    return computeDiffs(from.getValue(), to.getValue(), urn.toString(), auditStamp);
+  }
+
+  private List<ChangeEvent> computeDiffs(DataProcessInstanceRunEvent baseAspect,
+      DataProcessInstanceRunEvent targetAspect, String entityUrn, AuditStamp auditStamp) {
+    final DataProcessRunStatus baseStatus = getStatus(baseAspect);
+    final DataProcessRunStatus targetStatus = getStatus(targetAspect);
+
+    if (targetStatus != null && !targetStatus.equals(baseStatus)) {
+      String operationType = targetStatus.toString().equals("COMPLETE") ? "COMPLETED" : "STARTED";
+
+      return Collections.singletonList(ChangeEvent.builder()
```

Comment:
Nice - I really like how you're normalizing the DataJob and Assertion models into a single style. This keeps things nice and consistent.

Reply:
Thanks!